src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -330,12 +329,13 @@ public void periodic() {
       camera.updateInputs();
       camera.processInputs();
     }
-    odometryLock.lock(); // Prevents odometry updates while reading data
+    var odometryReadLock = PhoenixOdometryThread.getInstance().getReadLock();
```

Comment:
not sure how smelly this solution is, it is the lowest impact on the rest of the codebase though